src/theory/relevance_manager.h

```diff
@@ -81,6 +81,7 @@ class RelevanceManager : protected EnvObj
   using RlvPairHashFunction = PairHashFunction<Node, uint32_t, std::hash<Node>>;
   using NodeList = context::CDList<Node>;
   using NodeMap = context::CDHashMap<Node, Node>;
+  using NodeListMap = context::CDHashMap<Node, std::shared_ptr<NodeList>>;
```

Comment:
Why are we using a `std::shared_ptr` here?

Reply:
I'll change to unique.

Reply:
Actually, this is currently a limitation of our context-dependent data structures, we can't store unique_ptr due to the requirements on the templated classes.

Reply:
Ah I see...